subprojects/core/src/main/java/org/gradle/api/internal/artifacts/DefaultProjectComponentIdentifier.java

```diff
@@ -39,7 +41,7 @@ public DefaultProjectComponentIdentifier(BuildIdentifier buildIdentifier, Path i
     @Override
     public String getDisplayName() {
         String prefix;
-        if (identityPath == Path.ROOT) {
+        if (Objects.equals(identityPath, Path.ROOT)) {
```

Comment:
Alternatively, we could ensure that Path.ROOT is always deserialized correctly from CC (as the singleton Path.ROOT that is). Do you think that might still make sense?

Reply:
Yes.  That would be one less unnecessary instance.  I'd be interested in learning how to do this.

Do we have any way of noting in the code that a value like `Path.ROOT`, which (from reading the class) appears to be the only instance of a `Path` that will ever represent the path to a root project, can actually be duplicated by CC deserialization?  Is there any quick way to write a test to serialize/deserialize a root path and make sure the same instance is restored?